x-pack/plugins/licensing/server/license.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { createHash } from 'crypto';
+import { LicenseFeature } from './license_feature';
+import { LICENSE_STATUS, LICENSE_TYPE } from './constants';
+import { LicenseType, ILicense } from './types';
+
+function toLicenseType(minimumLicenseRequired: LICENSE_TYPE | string) {
+  if (typeof minimumLicenseRequired !== 'string') {
+    return minimumLicenseRequired;
+  }
+
+  if (!(minimumLicenseRequired in LICENSE_TYPE)) {
+    throw new Error(`${minimumLicenseRequired} is not a valid license type`);
+  }
+
+  return LICENSE_TYPE[minimumLicenseRequired as LicenseType];
+}
+
+export class License implements ILicense {
+  private readonly hasLicense: boolean;
+  private readonly license: any;
+  private readonly features: any;
+  private _signature!: string;
+  private objectified!: any;
+  private readonly featuresMap: Map<string, LicenseFeature>;
+
+  constructor(
+    license: any,
+    features: any,
+    private error: Error | null,
+    private clusterSource: string
+  ) {
+    this.hasLicense = Boolean(license);
+    this.license = license || {};
+    this.features = features;
+    this.featuresMap = new Map<string, LicenseFeature>();
+  }
+
+  public get uid() {
+    return this.license.uid;
+  }
+
+  public get status() {
+    return this.license.status;
+  }
+
+  public get isActive() {
```

Comment:
Legacy plugin defined it as a method. should either to follow the same interface or rename the key to avoid confusion?

Reply:
All of these are properties now.